flo-runner/src/main/java/com/spotify/flo/context/FloRunner.java

```diff
@@ -81,7 +85,16 @@ private static Config defaultConfig() {
    * @return a {@link Result} with the value and throwable (if thrown)
    */
   public static <T> Result<T> runTask(Task<T> task, Config config) {
-    return new Result<>(new FloRunner<T>(config).run(task));
+    return new Result<>(new FloRunner<T>(config).run(task), loadTerminationHooks());
+  }
+
+  private static Iterable<TerminationHook> loadTerminationHooks() {
+    final ServiceLoader<TerminationHookFactory> factories =
+        ServiceLoader.load(TerminationHookFactory.class);
+
+    return StreamSupport
```

Comment:
Maybe just do it in old school way: passing in `iterator` and later do `iterator.forEachRemaining`.

Reply:
nvm, it is a iterator of factories. makes sense to fail the hook instance creation early on.